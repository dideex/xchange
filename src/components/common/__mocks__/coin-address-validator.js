const WAValidator = () => ({
  validate: () => {},
  default: {
    validate: () => {},
  },
})

export default WAValidator
