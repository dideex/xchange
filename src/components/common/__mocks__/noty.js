class Noty {
  show() {}
}

Noty.__proto__.constructor = jest.fn()
export default Noty
