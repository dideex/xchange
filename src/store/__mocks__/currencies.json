{
  "data": [
    {
      "id": 0,
      "_id": "5bdc3a492fb62979e3105534",
      "base": "",
      "change": "0",
      "icon": "Bitcoin",
      "label": "BTC",
      "minimal": "0.1",
      "name": "Bitcoin",
      "order": 1,
      "price_usd": "4",
      "reserve": "10000",
      "source": "zzzZzzzzz",
      "type": "crypto"
    },
    {
      "id": 1,
      "_id": "5bdc3a492fb62979e3105532",
      "base": "",
      "change": "18.25",
      "icon": "Ethereum",
      "label": "ETH",
      "minimal": "1",
      "name": "Ethereum",
      "order": 2,
      "price_usd": "2",
      "reserve": "10000",
      "source": "1234 1234 1234 1234",
      "type": "crypto"
    },
    {
      "id": 2,
      "_id": "5bdc3a492fb62979e310553c",
      "base": "Ruble",
      "change": "0.00013913743704956245",
      "icon": "Sberbank",
      "label": "RUR",
      "mask": "____ ____ ____ ____",
      "minimal": "1000",
      "name": "Сбербанк руб",
      "order": 3,
      "price_usd": "1",
      "reserve": "1000000",
      "source": "4321 4321 4321 4321"
    }
  ],
  "userRate": 0.9
}
