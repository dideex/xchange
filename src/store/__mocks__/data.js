export const fakeLocale = 'fake-FK'
export const fakeToken = 'fakeToken'
export const fakeUser = {username: 'Batman', password: 'Nopassword'}
export const fakeUserData = {
  password: fakeUser.password,
  wallets: [{bitcoin: 1234}],
  lastOperations: [],
  username: fakeUser.username,
  email: fakeUser.username,
  login: fakeUser.username,
  convertedAmount: 1000,
}
