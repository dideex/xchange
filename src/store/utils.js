import Cookie from 'js-cookie'

export const setToken = (token, isAdmin) => {
  if (isAdmin) {
    Cookie.set('key', token)
    Cookie.set('isAdmin', true)
  } else Cookie.set('key', token)
}

export const logout = () => {
  Cookie.remove('key')
  Cookie.remove('isAdmin')
}

export const getToken = () => Cookie.get('key') || null
export const getAdminStatus = () => Cookie.get('isAdmin') || null
