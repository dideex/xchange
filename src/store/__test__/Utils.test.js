import Cookie from 'js-cookie'
import * as utils from '../utils'
import {fakeToken} from '../__mocks__/data'

jest.mock('js-cookie', () => ({
  set: () => {},
  get: () => {},
  remove: () => {},
}))

describe('Utils test', () => {
  it('Set token test', () => {
    Cookie.set = jest.fn()
    utils.setToken(fakeToken)
    expect(Cookie.set).toHaveBeenCalledTimes(1)
    expect(Cookie.set).toHaveBeenCalledWith('token', fakeToken)
  })

  it('Set admin token', () => {
    Cookie.set = jest.fn()
    utils.setToken(fakeToken, true)
    expect(Cookie.set).toHaveBeenCalledTimes(2)
    expect(Cookie.set).toHaveBeenCalledWith('token', fakeToken)
    expect(Cookie.set).toHaveBeenCalledWith('isAdmin', true)
  })

  it('Remove token', () => {
    Cookie.remove = jest.fn()
    utils.setToken(fakeToken)
    utils.logout()
    expect(Cookie.remove).toHaveBeenCalledTimes(2)
    expect(Cookie.remove).toHaveBeenCalledWith('token')
    expect(Cookie.remove).toHaveBeenCalledWith('isAdmin')
  })

  it('Get token', () => {
    Cookie.get = jest.fn(() => fakeToken)
    const res = utils.getToken()
    expect(Cookie.get).toHaveBeenCalledTimes(1)
    expect(Cookie.get).toHaveBeenCalledWith('token')
    expect(res).toMatchSnapshot()
  })

  it('Get empty token', () => {
    Cookie.get = jest.fn()
    const res = utils.getToken()
    expect(Cookie.get).toHaveBeenCalledTimes(1)
    expect(Cookie.get).toHaveBeenCalledWith('token')
    expect(res).toBeNull()
  })

  it('Get admin status', () => {
    Cookie.get = jest.fn(() => true)
    const res = utils.getAdminStatus()
    expect(Cookie.get).toHaveBeenCalledTimes(1)
    expect(Cookie.get).toHaveBeenCalledWith('isAdmin')
    expect(res).toBeTruthy()
  })

  it('Get empty admin status', () => {
    Cookie.get = jest.fn()
    const res = utils.getAdminStatus()
    expect(Cookie.get).toHaveBeenCalledTimes(1)
    expect(Cookie.get).toHaveBeenCalledWith('isAdmin')
    expect(res).toBeNull()
  })
})
