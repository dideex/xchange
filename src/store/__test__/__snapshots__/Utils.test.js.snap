// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Utils test Get token 1`] = `"fakeToken"`;
